project_management/pm/src/App.js

```diff
@@ -8,49 +8,44 @@ import Homepage from "./components/Homepage";
 import TopNav from "./components/TopNav";
 import TasksList from "./components/TaskFolder/TasksList";
 import Task from "./components/TaskFolder/Task";
-import TaskForm from "./components/TaskFolder/TaskForm";
 import ProjectsList from "./components/ProjectsFolder/ProjectsList";
-import ProjectForm from "./components/ProjectsFolder/ProjectForm";
 import Project from "./components/ProjectsFolder/Project";
-import Team from "./components/TeamsFolder/Teams";
-import TeamsForm from "./components/TeamsFolder/TeamsForm";
+import Team from "./components/TeamsFolder/Team";
 import TeamsList from "./components/TeamsFolder/TeamsList";
-import Cookies from 'universal-cookie';
-
-export const  cookiesContext = createContext();
-const cookies=new Cookies();
+import Cookies from "universal-cookie";
+import Profile from "./components/ProfileFolder/Profile";
+import "./App.css"
+export const cookiesContext = createContext();
+const cookies = new Cookies();
 
 function App() {
-
-
-
-
-
   return (
     <Router>
       <cookiesContext.Provider value={cookies}>
-      <TopNav  />
-      <Routes>
-        <Route path="/" element={<Homepage />} />
+        <TopNav />
+        <Routes>
 
-        <Route path="/login" element={<Login  />} />
-        <Route path="signup" element={<Signup />} />
+          <Route path="/" element={<Homepage />}/>
+          <Route path="/login" element={<Login />} />
+          <Route path="/signup" element={<Signup />} />
 
-        <Route path="/tasks" element={<TasksList />} />
-        <Route path="/tasks/:id" element={<Task />} />
-        <Route path="/addtask" element={<Task />} />
+          <Route path="/tasks" element={<TasksList />} />
+          <Route path="/tasks/:id" element={<Task />} />
+          <Route path="/addtask" element={<Task />} />
```

Comment:
make a page for 404 and one for unauthorized

what will happen if i try to open /tasks without login ?

Reply:
if you try to open / without login you will be redirected to /login

instead I want to use a middleware 

if you try to open tasks a page with no data will appear